lib/api/bucketPutTagging.js

```diff
@@ -58,6 +58,16 @@ function checkTags(tagSet, bucket, log, cb) {
         } else if (tags[i].Value.length > 256) {
             errMsg = tagValueInvalid;
         }
+        if (i > 50) {
+            log.debug(tooManyTags,
+                {
+                    method: 'bucketPutTagging',
+                    error: errors.BadRequest,
+                });
+            const error = errors.BadRequest.customizeDescription(
+                tooManyTags);
+            return cb(error);
+        }
```

Comment:
While in the end this will still be an improvement, when Anurag asked to put it before the other tests, he likely meant before the loop. Either way I'm sure it's not a big deal at this point.

Reply:
I checked, and AWS raises the first error it faces, so if I check for 50 tags before anything else we might end up showing "error more than 50 tags" before showing "invalid tag".

By doing this way, we check all errors on the first 50 tags and if there is no errors, but there is more than 50 tags, we raise the "more than 50 tags error"